### PR TITLE
Fix a bug in DiskTetMeshFacetedSurface (defined in tetmesh_faceted_surfaces.h) regarding the nodes of boundary elements

### PR DESCRIPTION
**Pull request by RupinderM**
## Problem

In `DiskTetMeshFacetedSurface`, we count the boundary nodes of an element on the outer boundary to check there are only two, but an element may have all three nodes on boundaries since there is an internal annular boundary. A similar issue arises when cycling nodes such that nodes of index 2 aren't on the boundary. 

## Solution

The above issues were fixed by checking that a node is on the outer boundary rather than on a boundary.

N.B. This patch only affects the `meshing/adaptive_tet_meshes` self-test, which has passed; see linked tests below.

## Self-tests

- [Ubuntu self-tests](https://github.com/RupinderM/oomph-lib/actions/workflows/self-tests-ubuntu.yaml)
- [macOS self-tests](https://github.com/RupinderM/oomph-lib/actions/workflows/self-tests-macos.yaml)

### CODE REVIEW

File: demo_drivers/meshing/adaptive_tet_meshes/tetmesh_faceted_surfaces.h [hunk 1]
```diff
@@ -506,13 +506,12 @@ class DiskTetMeshFacetedSurface : public virtual TetMeshFacetedSurface
...
          Node* nod_pt=el_pt->node_pt(j);
-         if (nod_pt->is_on_boundary()) count++;
+         if (nod_pt->is_on_boundary(b)) count++;
         }
        if (count==2)
         {
          is_on_boundary[el_pt]=true;
-         if ((el_pt->node_pt(0)->is_on_boundary())&&
-             (el_pt->node_pt(1)->is_on_boundary()))
+         if (!el_pt->node_pt(2)->is_on_boundary(b))
```
Comment (MatthiasHeilManchester) [inline demo_drivers/meshing/adaptive_tet_meshes/tetmesh_faceted_surfaces.h:514]: I'd put brackets around the thing that's negated, i.e. if (!(el_pt....))

File: demo_drivers/meshing/adaptive_tet_meshes/tetmesh_faceted_surfaces.h [hunk 2]
```diff
@@ -522,13 +521,13 @@ class DiskTetMeshFacetedSurface : public virtual TetMeshFacetedSurface
...
            Node* nod2_pt=el_pt->node_pt(2);
-           if (!el_pt->node_pt(0)->is_on_boundary())
+           if (!el_pt->node_pt(0)->is_on_boundary(b))
             {
              el_pt->node_pt(0)=nod1_pt;
              el_pt->node_pt(1)=nod2_pt;
              el_pt->node_pt(2)=nod0_pt;
             }
-           else if (!el_pt->node_pt(1)->is_on_boundary())
+           else if (!el_pt->node_pt(1)->is_on_boundary(b))
```
Comment (MatthiasHeilManchester) [inline demo_drivers/meshing/adaptive_tet_meshes/tetmesh_faceted_surfaces.h:530]: I'd put brackets around the thing that's negated, i.e. if (!(el_pt....))

**Review by MatthiasHeilManchester — CHANGES_REQUESTED** [review]
Thanks for spotting/fixing this. Three minor (and identical) tweaks

File: demo_drivers/meshing/adaptive_tet_meshes/tetmesh_faceted_surfaces.h (hunk 1 above)
Comment (RupinderM) [reply demo_drivers/meshing/adaptive_tet_meshes/tetmesh_faceted_surfaces.h]: Added in the latest commit, see the URL below for the changes applied.
https://github.com/oomph-lib/oomph-lib/pull/97/commits/6309f360261807c16f6d5b80452be2205c85237b

**Comment by MatthiasHeilManchester** [comment]
All lovely! Thanks for fixing this.

**Pull request merged**